### DISCUSSION TITLE
How to increase full text query speed using interval

### DISCUSSION THREAD

**Discussion by Mhrshdmh** [ClickHouse/ClickHouse#71708] [Q&A]
I’m encountering an issue with configuring full-text search in ClickHouse v23 for my operational database.
Goal: I want to enable efficient search functionality on a column named raw_event_msg in the event01 table, which contains raw log messages. My goal is to perform LIKE searches, e.g.,

SELECT count() FROM core.event01 WHERE raw_event_msg LIKE '%windows%';

This query currently takes around 166 seconds on 8.7 billion rows, which I’m hoping to improve.

I followed the full-text search configuration as recommended [here](https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/invertedindexes):

Enabled full-text index setting:
SET allow_experimental_full_text_index = 1;

Handled nullable column: Since raw_event_msg is Nullable(String), I created a new column raw_event_msg_nonnull with non-null values using:
ALTER TABLE core.event01
ADD COLUMN raw_event_msg_nonnull String MATERIALIZED ifNull(raw_event_msg, '');

Added inverted index:
ALTER TABLE core.event01 
ADD INDEX inverted_index_raw_event_msg raw_event_msg_nonnull TYPE inverted GRANULARITY 64;

However, when running the query with raw_event_msg_nonnull, e.g.,
SELECT count() FROM core.event01 WHERE raw_event_msg_nonnull LIKE '%windows%';

the performance is even slower than before applying the full-text configuration. I’ve tried adjusting GRANULARITY to different values, but it hasn’t improved the result.
----------
Question: Is there an issue with my configuration, or am I missing something critical in setting up full-text search? Any guidance on improving query performance with inverted indexes would be greatly appreciated.

Thanks for your help!

**Comment by leeningli** [comment]
the full text/inverted index feature is not released and had been removed from 2024 RoadMap
maybe need to wait it released

**Comment by dashtimehran** [comment]
I appreciate all the great master piece that you a working on. I wanted to know if it is possible to include full text feature in very next RoadMap of start of 2025? It is a pity that this good work does not have this feature?

**Comment by UnamedRus** [comment]
```
SELECT count() FROM core.event01 WHERE raw_event_msg LIKE '%windows%';
```

Did you checked tokenbf index before using reverse index one?
